prototype2/gdgem/NMXConfig.cpp

```diff
@@ -3,86 +3,89 @@
 #include <common/Trace.h>
 #include <gdgem/NMXConfig.h>
 
-#include <dataformats/multigrid/inc/json.h>
+#include <nlohmann/json.hpp>
 #include <fstream>
 #include <sstream>
 
 #undef TRC_LEVEL
 #define TRC_LEVEL TRC_L_DEB
 
 NMXConfig::NMXConfig(std::string jsonfile) {
-  Json::Value root{};
-  Json::Reader reader{};
+  nlohmann::json root;
+
   std::ifstream t(jsonfile);
-  std::string str((std::istreambuf_iterator<char>(t)),
+  std::string jsonstring((std::istreambuf_iterator<char>(t)),
                   std::istreambuf_iterator<char>());
 
-  if (!reader.parse(str, root, 0)) {
+  try {
+    root = nlohmann::json::parse(jsonstring);
+  }
+  catch (...) {
```

Comment:
It is not a huge problem in my opinion that you are using `catch (...)` as you are executing only one line. However, according to the documentation, `nlohmann::json::parse` only throws one type of error (`nlohmann::json::parse_error`) and thus catching this error would give you more information about the error by printing it (using the `what()`) member function.

Reply:
Hmm. This was just copy paste from another example. But is there any guarantee 
that the library will not change behaviour at some time and potentially throw other
exceptions? In which case ..., which I assume means catch all exceptions, is more guarded
against future changes in behaviour?

Reply:
I would consider it very bad form of the library developer to change the type of the exception that is thrown but there are at least two different options if you are worried about this:

1. Most exceptions use `std::exception` as the "root" (?) class, thus you can do this:
```c++
try {
  //parsing code
} catch (std::exception &e) {
  printf(e.what());
}
```
2. It is possible to use multiple `catch()` statements:
```c++
try {
  // Parsing code
} catch (json::parse_error &e) {
 printf(e.what());
} catch (...) {
// Do something else.
}
```

Reply:
As is is not a huge problem, and because I don't see the advantage of making it more complicated, I will probably let this stay.